source/core_http_client.c

```diff
@@ -574,10 +577,23 @@ static int8_t caseInsensitiveStringCmp( const char * str1,
                                         size_t n )
 {
     size_t i = 0U;
+    /* Inclusion of inbetween variables for coverity rule 13.2 compliance */
+    int32_t firstChar;
+    int32_t secondChar;
 
     for( i = 0U; i < n; i++ )
     {
-        if( toupper( str1[ i ] ) != toupper( str2[ i ] ) )
+        /* MISRA Ref 21.13.1 [Essential type casting] */
```

Comment:
This whole change is not necessary. You just need to add ( int ) casts to the toupper arguments.
```suggestion
        if( toupper( ( int ) str1[ i ] ) != toupper( ( int ) str2[ i ] ) )
```

Reply:
There were a couple comments in relation to this function so putting this comment here for future reference: It's likely an issue with the static analysis that says this line is in error. We've still decided to use inbetween variables to remove the violation. The violation can be removed by only using one additional variable, allowing the line to pass with only declaring a firstChar variable and using toupper directly on the second argument. That said, I thought it looked a little messy to declare a variable for only one of the arguments.